src/EnergyPlus/FaultsManager.cc

```diff
@@ -206,406 +276,700 @@ namespace FaultsManager {
+		Real64 OffsetAct( 0.0 ); // actual offset after applying the modification factor
+		
+		// Check fault availability schedules
+		if ( GetCurrentScheduleValue( this->AvaiSchedPtr ) > 0.0 ) {
+		
+			// Check fault severity schedules 
+			if ( this->SeveritySchedPtr >= 0 ) {
+				FaultFac = GetCurrentScheduleValue( this->SeveritySchedPtr );
+			} else {
+				FaultFac = 1.0;
+			}
+		}
+		
+		OffsetAct = FaultFac * this->Offset;
+
+		return OffsetAct;
+	}
+
+	void
+	FaultPropertiesChillerSWT::CalFaultChillerSWT(
+		bool FlagVariableFlow, // True if chiller is variable flow and false if it is constant flow
+		Real64 FaultyChillerSWTOffset, // Faulty chiller SWT sensor offset
+		Real64 Cp, // Local fluid specific heat
+		Real64 EvapInletTemp, // Chiller evaporator inlet water temperature 
+		Real64 & EvapOutletTemp, // Chiller evaporator outlet water temperature 
+		Real64 & EvapMassFlowRate, // Chiller mass flow rate
+		Real64 & QEvaporator // Chiller evaporator heat transfer rate
 	)
 	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Rongpeng Zhang
+		//       DATE WRITTEN   Jun. 2016
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// To calculate the mass flow rate and supply water temperature of a chiller with faulty SWT sensor.
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		// Variables for fault free cases
+		Real64 EvapOutletTemp_ff = EvapOutletTemp;   // Chiller supply water temperature, fault free [C]
+		Real64 EvapMassFlowRate_ff = EvapMassFlowRate; // Chiller mass flow rate, fault free [kg/s]
+		Real64 QEvaporator_ff = QEvaporator; // Chiller evaporator heat transfer rate, fault free [W]
+		
+		// Variables for faulty cases
+		Real64 EvapOutletTemp_f = EvapOutletTemp_ff;    // Chiller supply water temperature, faulty case [C]
+		Real64 EvapMassFlowRate_f = EvapMassFlowRate_ff;  // Chiller mass flow rate, faulty case [kg/s]
+		Real64 QEvaporator_f = QEvaporator_ff;  // Chiller evaporator heat transfer rate, faulty case [W]
+		
+		if( !FlagVariableFlow ){
+		// Chillers with ConstantFlow mode
+		
+			EvapOutletTemp_f = EvapOutletTemp_ff - FaultyChillerSWTOffset;
+			
+			if( ( EvapInletTemp > EvapOutletTemp_f ) && ( EvapMassFlowRate_ff > 0 )){
+				QEvaporator_f = EvapMassFlowRate_ff * Cp * ( EvapInletTemp - EvapOutletTemp_f );
+			} else {
+				EvapMassFlowRate_f = 0.0;
+				QEvaporator_f = 0.0;
+			}
+		
+		} else {
+		// Chillers with LeavingSetpointModulated mode
+		
+			EvapOutletTemp_f = EvapOutletTemp_ff - FaultyChillerSWTOffset;
+			
+			if( ( EvapInletTemp > EvapOutletTemp_f ) && ( Cp > 0 ) && ( EvapMassFlowRate_ff > 0 ) ){
+				EvapMassFlowRate_f = QEvaporator_ff / Cp / ( EvapInletTemp - EvapOutletTemp_ff );
+				QEvaporator_f =  EvapMassFlowRate_f * Cp * ( EvapInletTemp - EvapOutletTemp_f );
+			} else {
+				EvapMassFlowRate_f = 0.0;
+				QEvaporator_f = 0.0;
+			}
+		}
+		
+		// Return variables
+		EvapOutletTemp = EvapOutletTemp_f;
+		EvapMassFlowRate = EvapMassFlowRate_f;
+		QEvaporator = QEvaporator_f;
+	}
```

Comment:
These codes [e.g., ChillerReformulatedEIR.cc, Line 2276] should execute before calling to CalFaultChillerSWT [e.g., ChillerReformulatedEIR.cc, Line 2288], so the load (QEvaporator) will not be reverted back. It's a mistake in the  "ChillerElectricEIR.cc". I will double check this for all chiller models.
I checked the following key parameters at the end of CalcReformEIRChillerModel function and find the values are what we expect: (1) EvapMassFlowRate (2) EvapOutletTemp & EvapDeltaTemp (3) QEvaporator